Core/src/AmbiguityResolution/AthenaAmbiguityResolution.cpp

```diff
@@ -0,0 +1,204 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2024 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "Acts/AmbiguityResolution/AthenaAmbiguityResolution.hpp"
+
+#include "Acts/EventData/Measurement.hpp"
+#include "Acts/EventData/SourceLink.hpp"
+
+#include <stdexcept>
+
+std::vector<std::size_t> Acts::AthenaAmbiguityResolution::getCleanedOutTracks(
+    std::vector<double> trackScore,
+    std::vector<std::map<std::size_t, TrackFeatures>>& counterMaps,
+    std::vector<std::vector<std::tuple<std::size_t, std::size_t, bool>>>
+        measurementsPerTrack) const {
+  std::vector<std::size_t> cleanTracks;
+
+  ACTS_VERBOSE("Cleaning tracks");
+
+  if (trackScore.size() != measurementsPerTrack.size()) {
+    throw std::invalid_argument(
+        "Track score and measurementsPerTrack size mismatch");
+  }
+
+  std::size_t numberOfTracks = measurementsPerTrack.size();
+  ACTS_DEBUG("Number of tracks: " << numberOfTracks);
+
+  boost::container::flat_map<std::size_t,
```

Comment:
I would actually suggest a simple `std::vector<int> tracksPerMeasurement (nMeasurements, 0);`

and then, when looping on the tracks you simply increment the corresponding element (i.e. `++tracksPerMeasurement[iMeasurement];`)

In the end you simply check the size of the sets. So an integer will do the same

Reply:
I took this implementation from greedy solver.